plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/panel/GitPanelPresenter.java

```diff
@@ -0,0 +1,201 @@
+    this.view = view;
+    this.service = service;
+    this.changesPanelPresenter = changesPanelPresenter;
+    this.appContext = appContext;
+    this.notificationManager = notificationManager;
+    this.gitResources = gitResources;
+    this.locale = locale;
+
+    this.view.setDelegate(this);
+    this.view.setChangesPanelView(this.changesPanelPresenter.getView());
+
+    this.currentProject = appContext.getRootProject();
+    this.alteredFiles = new MutableAlteredFiles(currentProject);
+
+    if (partStack == null || !partStack.containsPart(this)) {
+      workspaceAgent.openPart(this, PartStackType.NAVIGATION);
+    }
+
+    subscribeToGitEvents.addSubscriber(this);
+  }
+
+  /** Invoked each time when panel is activated. */
+  @Override
+  public void onOpen() {
+    // Update file list according to project explorer selection
+    Project selectedProject = appContext.getRootProject();
+
+    if (selectedProject == null) {
+      updateChangedFiles(null);
+      return;
+    } else if (selectedProject != currentProject) {
+      currentProject = selectedProject;
+      reloadChangedFilesList();
+    }
+  }
+
+  private void reloadChangedFilesList() {
+    service
+        .diff(currentProject.getLocation(), null, NAME_STATUS, false, 0, REVISION, false)
+        .then(
+            diff -> {
+              alteredFiles = new MutableAlteredFiles(currentProject, diff);
+              updateChangedFiles(alteredFiles);
+            })
+        .catchError(
+            arg -> {
+              notificationManager.notify(locale.diffFailed(), FAIL, NOT_EMERGE_MODE);
+            });
+  }
+
+  private void updateChangedFiles(AlteredFiles alteredFiles) {
+    changesPanelPresenter.show(alteredFiles);
+  }
+
+  @Override
+  public String getTitle() {
+    return locale.panelTitle();
+  }
+
+  @Override
+  public SVGResource getTitleImage() {
+    return gitResources.git();
+  }
+
+  @Override
+  public IsWidget getView() {
+    return view;
+  }
+
+  @Override
+  public String getTitleToolTip() {
+    return locale.panelTitleToolTip();
+  }
+
+  @Override
+  public void go(AcceptsOneWidget container) {
+    container.setWidget(view);
+  }
+
+  public void show() {
+    onActivate();
+  }
+
+  public void hide() {
+    partStack.minimize();
+  }
+
+  public boolean isOpened() {
+    return partStack.getActivePart() == this;
+  }
+
+  @Override
+  public void onFileChanged(String endpointId, FileChangedEventDto dto) {
+    switch (dto.getStatus()) {
+      case ADDED:
+        // do nothing
+        break;
+      case MODIFIED:
+        if (alteredFiles.addFile(removeProjectName(dto.getPath()), Status.MODIFIED)) {
+          updateChangedFiles(alteredFiles);
+        }
+        break;
+      case UNTRACKED:
+        // do nothing
+        break;
+      case NOT_MODIFIED:
+        if (alteredFiles.removeFile(removeProjectName(dto.getPath()))) {
+          updateChangedFiles(alteredFiles);
+        }
+        break;
```

Comment:
add default

Reply:
All cases are present, no need to add default.